framework/Samplers/EnsembleForward.py

```diff
@@ -185,7 +186,11 @@ def localInitialize(self):
         self.instanciatedSamplers[samplingStrategy].localGenerateInput(None,None)
         self.instanciatedSamplers[samplingStrategy].inputInfo['prefix'] = self.instanciatedSamplers[samplingStrategy].counter
         self.samplersCombinations[samplingStrategy].append(copy.deepcopy(self.instanciatedSamplers[samplingStrategy].inputInfo))
+        mKeys, mParams = self.instanciatedSamplers[samplingStrategy].provideExpectedMetaKeys()
+        metadataKeys.extend(mKeys)
+        mParams.update(mParams)
```

Comment:
Do you mean **metadataParams.update(mParams)**?

Reply:
the mParams is within the for and while loop, and it was reseted during each iteration. It seems to me that you can not use it directly to store mParams for all the iterations. Please correct me.

Reply:
I modified it and put it outside